stew/objects.nim

```diff
@@ -59,13 +59,15 @@ when not compiles(len((1, 2))):
 # Get an object's base type, as a cstring. Ref objects will have an ":ObjectType"
 # suffix.
 # From: https://gist.github.com/stefantalpalaru/82dc71bb547d6f9178b916e3ed5b527d
-proc baseType*(obj: RootObj): cstring =
+proc baseType*(obj: RootObj): cstring {.deprecated.} =
   when not defined(nimTypeNames):
     raiseAssert("you need to compile this with '-d:nimTypeNames'")
+  when defined(arc) or defined(orc):
+    raiseAssert("baseType doesn't work with arc / orc")
```

Comment:
why not compiletime?

Reply:
It's always compiled since it's not generic

Reply:
well yes, so that's the point: it should fail compilation when any of these are enabled

Reply:
the whole function should not exist (similar to shallowCopy)

Reply:
Just `nim c stew/objects` fails with `{.error.}` (even without actually using the function)

Reply:
```nim
when defined(gcArc) or defined(gcOrc):
  proc baseType*(obj: RootObj): cstring {.error: "baseType is not available in ARC/ORC".}
else:
  proc baseType*(obj: RootObj): cstring =
```
Works for me.

Reply:
what? `when not defined(orc/arc): proc baseType(...)` - there's no `{.error.}` here.. we should really change the `nimTypeNames` runtime assert to compile-time as well, they're the same

Reply:
It works if I switch to `proc baseType*[T: RootObj](obj: T): cstring {.deprecated.} =` (since the error will be done at instantiation), but not sure we want do to that

Reply:
Switched to generic, why not

Reply:
Is there something wrong with https://github.com/status-im/nim-stew/pull/148#discussion_r1017974909? It breaks at compile time and gives a nicer messages for ARC/ORC.

```
Error: baseType is not available in ARC/ORC; usage of 'baseType' is an {.error.} defined at
```

Reply:
You can also include `nimTypeNames` for defines and gives a proper error message.

Reply:
Sorry github (or my brain) derped and didn't show your comment. Better indeed, thanks